### PR TITLE
Adds a GitLab ci file

### PR DESCRIPTION
**Pull request by fhaynes**

### CODE REVIEW

**Comment by fhaynes** [comment]
bors try

File: bors.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Test suite must pass on both Travis and AppVeyor before merging into `master`.
```
Comment (jojolepro) [inline bors.toml]: Outdated doc

File: bors.toml [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
 status = [
-    "continuous-integration/travis-ci/push"
+    "continuous-integration/travis-ci/push",
+    'ci/gitlab/%',
```
Comment (jojolepro) [inline bors.toml]: :zzz: next time we try to merge anything.

Comment (fhaynes) [reply bors.toml]: Fixed

Comment (fhaynes) [reply bors.toml]: It completes in about 30 minutes on GitLab.

File: bors.toml (hunk 1 above)
Comment (fhaynes) [reply bors.toml]: Updated.

**Pull request merged**